### PR TITLE
support for virtual device app automation and attach to w3c sessions

### PR DESCRIPTION
**Pull request by shiva-guntoju**

### CODE REVIEW

File: app/renderer/actions/Session.js [hunk 1]
```diff
@@ -347,15 +347,19 @@ export function newSession(caps, attachSessId = null) {
         username = session.server.lambdatest.username || process.env.LAMBDATEST_USERNAME;
         if (desiredCapabilities.hasOwnProperty.call(desiredCapabilities, 'lt:options')) {
           desiredCapabilities['lt:options'].source = 'appiumdesktop';
-          desiredCapabilities['lt:options'].isRealMobile = true;
+          if (desiredCapabilities['lt:options'].isRealMobile === undefined || desiredCapabilities['lt:options'].isRealMobile){
```
Comment (rohithmone-lt) [inline app/renderer/actions/Session.js]: Missing space before opening brace

File: app/renderer/actions/Session.js [hunk 2]
```diff
@@ -347,15 +347,19 @@ export function newSession(caps, attachSessId = null) {
...
+          }
           if (session.server.advanced.useProxy) {
             desiredCapabilities['lt:options'].proxyUrl = isUndefined(session.server.advanced.proxy)
               ? ''
               : session.server.advanced.proxy;
           }
         } else {
           desiredCapabilities['lambdatest:source'] = 'appiumdesktop';
-          desiredCapabilities['lambdatest:isRealMobile'] = true;
+          if (desiredCapabilities['lambdatest:isRealMobile'] === undefined || desiredCapabilities['lambdatest:isRealMobile']){
```
Comment (rohithmone-lt) [inline app/renderer/actions/Session.js]: Missing space before opening brace

Comment (shiva-guntoju) [reply app/renderer/actions/Session.js]: done

File: app/renderer/actions/Session.js (hunk 1 above)
Comment (shiva-guntoju) [reply app/renderer/actions/Session.js]: done

File: app/renderer/actions/Session.js [hunk 3]
```diff
@@ -572,6 +576,9 @@ export function newSession(caps, attachSessId = null) {
         let attachedSessionCaps = {};
         if (attachedSession) {
           attachedSessionCaps = attachedSession.capabilities;
+          if (attachedSessionCaps.platformName === undefined) {
+            attachedSessionCaps = attachedSession.capabilities.capabilities;
```
Comment (rohithmone-lt) [inline app/renderer/actions/Session.js:586]: just verify with w3c:false and w3c:true for this change and verify everything related to appium inspector for real devices

Comment (rohithmone-lt) [reply app/renderer/actions/Session.js:580]: and verify everything related to appium inspector for real devices

**Review by rohithmone-lt — APPROVED** [review]

File: app/renderer/actions/Session.js (hunk 3 above)
Comment (rohithmone-lt) [reply app/renderer/actions/Session.js:580]: done

**Review by rohithmone-lt — APPROVED** [review]

**Review by swngarg-lt — APPROVED** [review]

**Review by anurag-lambdatest — APPROVED** [review]

**Pull request merged**